content/en/blog/2019/isolated-clusters/index.md

```diff
@@ -0,0 +1,1878 @@
+    metadata:
+      name: reviews-reader
+    spec:
+      subjects:
+      - user: "cluster.local/ns/istio-private-gateways/sa/istio-private-ingressgateway-service-account"
+      roleRef:
+        kind: ServiceRole
+        name: reviews-reader
+    ---
+    apiVersion: rbac.istio.io/v1alpha1
+    kind: ServiceRoleBinding
+    metadata:
+      name: ratings-reader
+    spec:
+      subjects:
+      - user: "cluster.local/ns/bookinfo/sa/bookinfo-reviews"
+      - user: "cluster.local/ns/istio-private-gateways/sa/istio-private-ingressgateway-service-account"
+      roleRef:
+        kind: ServiceRole
+        name: ratings-reader
+    EOF
+    {{< /text >}}
+
+1.  Enable [Istio RBAC](/docs/concepts/security/#authorization) on the `bookinfo` namespace.
+
+    {{< warning >}}
+    If you already enabled Istio RBAC on some of your namespaces, add `bookinfo` to the list of the included namespaces.
+    The command below assumes that `bookinfo` is the only namespace you enabled RBAC on.
+    {{< /warning >}}
+
+    {{< text bash >}}
+    $ kubectl apply --context=$CTX_CLUSTER2 -f - <<EOF
+    apiVersion: "rbac.istio.io/v1alpha1"
+    kind: ClusterRbacConfig
+    metadata:
+      name: default
+      namespace: istio-system
+    spec:
+      mode: ON_WITH_INCLUSION
+      inclusion:
+        namespaces: [ bookinfo ]
+    EOF
+    {{< /text >}}
+
+1.  Access the webpage of the Bookinfo application in `cluster1` and verify that the application continues to work.
+    It means that the authorized access is allowed and you configured your policy rules correctly.
+
+1.  Check that unauthorized access is denied. Send a GET request from `ratings` to `reviews`:
+
+    {{< text bash >}}
+    $  kubectl exec -it $(kubectl get pod -l app=ratings --context=$CTX_CLUSTER2 -n bookinfo -o jsonpath='{.items[0].metadata.name}') --context=$CTX_CLUSTER2 -n bookinfo -c ratings -- curl myreviews:9080/reviews/0 -w "\nResponse code: %{http_code}\n"
+    RBAC: access denied
+    Response code: 403
+    {{< /text >}}
+
+### Enable RBAC on the ingress gateway
+
+1.  Currently, the only identities Istio RBAC can handle are the identities of Istio mutual TLS. In this case, you want
+    to control the traffic for non-Istio mutual TLS from external clusters. To do that, you can use Envoy's filter [`envoy.filters.http.rbac`](https://www.envoyproxy.io/docs/envoy/latest/configuration/http_filters/rbac_filter).
+    To instruct Istio to use that filter, create an
+    [`EnvoyFilter`](/docs/reference/config/networking/v1alpha3/envoy-filter/):
+
+    {{< text bash >}}
+    $ kubectl apply  --context=$CTX_CLUSTER2 -n istio-private-gateways -f - <<EOF
+    apiVersion: networking.istio.io/v1alpha3
+    kind: EnvoyFilter
+    metadata:
+      name: private-ingress-rbac
+    spec:
+      workloadLabels:
+        istio: private-ingressgateway
+      filters:
+      - filterName: envoy.filters.http.rbac
+        filterType: HTTP
+        listenerMatch:
+          portNumber: 15443
+  
```

Comment:
some of the process could clearly benefit from automation. The term "Mesh Federation", and possible support for it, is currently being discussed by the Istio community. Hopefully automation for a pattern, or patterns, similar the one I showed here will be provided by Istio in the not too distant future.